Fakebook.Posts/Fakebook.Posts.DataAccess/Repositories/PostsRepository.cs

```diff
@@ -206,5 +206,13 @@ public async Task<bool> UnlikeCommentAsync(int commentId, string userEmail)
             }
             return false;
         }
+
+        public async ValueTask<Post> GetPostAsync(int id)
+        {
+
+            var post = await _context.Posts.Where(b => b.Id == id).Select(b => b.ToDomain()).SingleOrDefaultAsync();
```

Comment:
the select call can't be converted to a sql call, so i think this may be loading every single post into memory.
example alternative: `(await _context.Posts.FirstOrDefaultAsync(b => b.Id == id)).ToDomain()`.
or, if Id is the primary key (i forget), then even `await _context.Posts.FindAsync(id)`.